ipaserver/install/replication.py

```diff
@@ -600,7 +600,18 @@ def finalize_replica_config(self, r_hostname, r_binddn=None,
             r_conn.simple_bind(r_binddn, r_bindpw)
         else:
             r_conn.gssapi_bind()
-        self._finalize_replica_settings(r_conn)
+        try:
+            self._finalize_replica_settings(r_conn)
+        except errors.DatabaseError as e:
+            # On FreeIPA < 3.3 masters lacking support for the attributes
+            # defined in REPLICA_FINAL_SETTINGS,
+            # the update will return Unwilling to perform
+            # Ignore the error
+            if str(e).startswith('Server is unwilling to perform'):
+                logger.debug("replication attribute not supported "
+                             "on remote master (%s)", e)
+            else:
+                raise e
```

Comment:
The changes looks okay. 
An other option is to apply the final settings (in _finalize_replica_settings) only  if the 'vendorVersion' >= 1.3* (search -b "" -s base). If version >= 1.3* and we get err=53, then it is a real issue.

Reply:
I'd prefer @tbordaz suggestion and check the root DSE version information:

```
>>> dse = api.Backend.ldap2.conn.read_rootdse_s()
>>> vendor_version_str = dse['vendorVersion'][0].decode('utf-8')
>>> vendor_version_str
'389-Directory/1.4.0.13 B2018.201.1334'
>>> mo = re.search('(\d+)\.(\d+)\.(\d+)[\.\d]*', vendor_version_str)
>>> vendor_version = tuple(int(v) for v in mo.groups())
>>> vendor_version
(1, 4, 0)
```